### PR TITLE
tp: Add intervals_create C++ Perfetto SQL function

### PR DESCRIPTION
**Pull request by aMayzner**
Replace the O(n*m) SQL-based `create_slices` implementation with an O(n+m) C++ two-pointer algorithm. The structured query generator now calls `_intervals_create!` macro (backed by `__intrinsic_interval_create`) instead
of inlining a (inefficient) subquery.

## Changes

- Add `SortedTimestamps` type and `__intrinsic_timestamp_set_agg` aggregate
  to collect and sort timestamps from SQL into a C++ vector
- Add `__intrinsic_interval_create` scalar function that takes two
  `SortedTimestamps` pointers and produces a dataframe with `ts`/`dur` columns
- Add `_intervals_create!` PerfettoSQL macro in `intervals/create_slices.sql`
  that wires the aggregate + scalar function together
- Update `StructuredQueryGenerator::CreateSlices` to emit the macro call
  and register the module dependency
- Add 8 diff tests covering basic, interleaved, partial match, empty,
  equal-timestamp, unsorted, and no-match scenarios

### CODE REVIEW

**Comment by aMayzner** [comment]
> Having something which is called "slices" in a module called "intervals" is confusing and a layering violation:
> 
> * intervals = ts + dur
> * slices = ts + dur + name
> 
> Please fix the layering here. Either put it in slices module if you're creaitng slicdes or call it _create_intervals if you're creating intervals.

Done. I renamed everything except data explorer/sq to create intervals.

File: src/trace_processor/perfetto_sql/stdlib/intervals/create_intervals.sql [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+-- Example:
+-- ```
+-- SELECT * FROM _create_intervals!(
+--   (SELECT ts FROM starts_table),
+--   (SELECT ts FROM ends_table),
+--   ts,
+--   ts
+-- )
+-- ```
+CREATE PERFETTO MACRO _create_intervals(
```
Comment (LalitMaganti) [inline src/trace_processor/perfetto_sql/stdlib/intervals/create_intervals.sql]: _interval_create to match the pattern by _interval_intersect.

File: src/trace_processor/perfetto_sql/stdlib/intervals/create_intervals.sql [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    -- Table or subquery containing start timestamps.
+    starts_table TableOrSubquery,
+    -- Table or subquery containing end timestamps.
+    ends_table TableOrSubquery,
+    -- Name of the timestamp column in the starts table.
```
Comment (LalitMaganti) [inline src/trace_processor/perfetto_sql/stdlib/intervals/create_intervals.sql]: Can we please rename the input instead and hardcode ts as the column. This approach doesn't scale.

File: src/trace_processor/perfetto_sql/intrinsics/functions/type_builders.cc [hunk 3]
```diff
@@ -435,6 +436,37 @@ struct IntervalTreeIntervalsAgg
...
+    data.timestamps.push_back(sqlite::value::Int64(argv[0]));
+  }
+
+  static void Final(sqlite3_context* ctx) {
+    auto raw_agg_ctx = AggCtx::GetContextOrNullForFinal(ctx);
+    if (!raw_agg_ctx) {
+      return sqlite::result::Null(ctx);
+    }
+    auto& ts = raw_agg_ctx.get()->data.timestamps;
+    std::sort(ts.begin(), ts.end());
```
Comment (LalitMaganti) [inline src/trace_processor/perfetto_sql/intrinsics/functions/type_builders.cc]: Why are we sorting here instead of asking SQL to sort, eliding the sort if the data is already sorted?

File: src/trace_processor/perfetto_sql/intrinsics/functions/create_intervals.cc [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    auto* ends = sqlite::value::Pointer<SortedTimestamps>(
+        argv[1], SortedTimestamps::kName);
+
+    std::vector<std::string> col_names{"ts", "dur"};
+    std::vector<ColType> col_types{ColType::kInt64, ColType::kInt64};
+
+    dataframe::AdhocDataframeBuilder builder(
+        col_names, GetUserData(ctx)->pool,
+        dataframe::AdhocDataframeBuilder::Options{
+            col_types, dataframe::NullabilityType::kSparseNullWithPopcount});
```
Comment (LalitMaganti) [inline src/trace_processor/perfetto_sql/intrinsics/functions/create_intervals.cc:72]: Generally always prefer DenseNull for functions returning dataframes.

**Review by LalitMaganti — APPROVED** [review]

File: src/trace_processor/perfetto_sql/intrinsics/functions/create_intervals.cc (hunk 4 above)
Comment (aMayzner) [reply src/trace_processor/perfetto_sql/intrinsics/functions/create_intervals.cc:72]: Done
